src/DSi.cpp

```diff
@@ -918,42 +1008,92 @@ void ARM9Write8(u32 addr, u8 val)
     switch (addr & 0xFF000000)
     {
     case 0x03000000:
+    case 0x03800000:
         if (addr >= NWRAMStart[0][0] && addr < NWRAMEnd[0][0])
         {
-            u8* ptr = NWRAMMap_A[0][(addr >> 16) & NWRAMMask[0][0]];
-            if (ptr)
+            // Write to a bank is special, as it writes to all
+            // parts that are mapped and not just the highest priority
+            // See http://melonds.kuribo64.net/board/thread.php?pid=3974#3974
+            // so we need to iterate through all parts and write to all mapped here
+            u32 destPart = (addr >> 15) & NWRAMMask[0][0];
+            for (int page = 0; page < 4; page++)
             {
-                *(u8*)&ptr[addr & 0xFFFF] = val;
+                u8 bankInfo = (MBK[0][0 + (page / 4)] >> ((page % 4) * 8)) & 0xff;
+                if (!(bankInfo & 0x80))
+                    continue;
+                if ((bankInfo & 0x03) != 0)
+                    continue;
+                if (((bankInfo >> 3) & 0x03) != (destPart & 0x03))
+                    continue;
+                u8* ptr = &NWRAM_A[page * 0x8000];
+                *(u8*)&ptr[addr & 0x7FFF] = val;
+#if DSI_NWRAM_DEBUGOUT
+                printf("Set A9: Wrote %08x to NWRAM at %08x (source=%08x)\n", val, addr, ptr);
+                fflush(stdout);
+#endif
 #ifdef JIT_ENABLED
-                ARMJIT::CheckAndInvalidate<0, ARMJIT_Memory::memregion_NewSharedWRAM_A>(addr);
+                ARMJIT::CheckAndInvalidate<1, ARMJIT_Memory::memregion_NewSharedWRAM_A>(addr);
```

Comment:
why did you change that parameter? 0=ARM9

Reply:
Thanks. This is a c&p error I did not spot.